src/app/public/src/modules/shared/route.service.ts

```diff
@@ -126,36 +126,55 @@ export class StacheRouteService {
   }
 
   private sortRoutes(routes: StacheNavLink[]): StacheNavLink[] {
-
-    const sortByName = (a: any, b: any): number  => {
-      if (a.name.toLowerCase() < b.name.toLowerCase()) {
-        return -1;
-      } else if (a.name.toLowerCase() > b.name.toLowerCase()) {
-        return 1;
-      } else {
-        return 0;
-      }
-    };
-
-    const sortByOrder = (a: any, b: any): number => {
-      if (a.order < b.order) {
-        return -1;
-      } else if (a.order > b.order) {
-        return 1;
+    const sortedRoutes = routes.map(this.validateNavOrder)
+      .filter(route => !route.hasOwnProperty('order'))
+      .sort(this.sortByName);
+
+    const routesWithNavOrder = routes.map(this.validateNavOrder)
+      .filter(route => route.hasOwnProperty('order'))
+      .sort(this.sortByName)
+      .sort(this.sortByOrder);
+
+    routesWithNavOrder.forEach(route => {
+      let newIdx = route.order - 1;
+      if (newIdx < sortedRoutes.length) {
+        while (newIdx < sortedRoutes.length && sortedRoutes[newIdx].order <= route.order) {
+          newIdx++;
+        }
+        sortedRoutes.splice(newIdx, 0, route);
       } else {
-        return 0;
+        sortedRoutes.push(route);
       }
-    };
+    });
 
-    const orderedRoutes = routes.filter(route => route.hasOwnProperty('order'))
-      .sort(sortByName)
-      .sort(sortByOrder);
+    return sortedRoutes;
+  }
 
-    const unorderedRoutes = routes.filter(route => !route.hasOwnProperty('order'))
-      .sort(sortByName);
+  private validateNavOrder = (route: any) => {
```

Comment:
Mind formatting these methods in the "typescripty" way? Instead of using fat arrows...